From fb8846f85d0aa19d1f2428bb7fc63d6bbf845733 Mon Sep 17 00:00:00 2001
From: Kefu Chai <tchaikov@gmail.com>
Date: Wed, 9 Oct 2024 15:15:35 +0800
Subject: [PATCH] Address build failures when using Tip-of-Tree clang. (#4187)

When using ToT clang to build fmtlib, it complains 'sv' is not
initialized by a constant expression. This patch addresses this
issue.

this change is a backport of 22701d5f63fd9ba3ffa35fe94585b5bfcb69238b
---
 include/fmt/base.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/fmt/base.h b/include/fmt/base.h
index 62764942..819f0f6e 100644
--- a/include/fmt/base.h
+++ b/include/fmt/base.h
@@ -2877,11 +2877,12 @@ template <typename Char, typename... Args> class basic_format_string {
              std::is_reference<Args>::value)...>() == 0,
         "passing views as lvalues is disallowed");
 #if FMT_USE_CONSTEVAL
+    auto sv = string_view(str_);
     if constexpr (detail::count_named_args<Args...>() ==
                   detail::count_statically_named_args<Args...>()) {
       using checker =
           detail::format_string_checker<Char, remove_cvref_t<Args>...>;
-      detail::parse_format_string<true>(str_, checker(s));
+      detail::parse_format_string<true>(sv, checker(sv));
     }
 #else
     detail::check_format_string<Args...>(s);
-- 
2.46.2

